src/app/authentication/repo/user_impl.go

```diff
@@ -1,61 +1,95 @@
 package repo
-
 import (
 	"Stock_broker_application/constants"
 	"Stock_broker_application/models"
-	"database/sql"
-)
 
+	"gorm.io/gorm"
+)
 // UserRepositoryImpl is the implementation of UserRepository
 type UserRepositoryImpl struct {
-    db *sql.DB
+	db *gorm.DB
 }
 
 // NewUserRepositoryImpl creates a new instance of UserRepositoryImpl
-func NewUserRepositoryImpl(db *sql.DB) *UserRepositoryImpl {
-    return &UserRepositoryImpl{db: db}
+func NewUserRepositoryImpl(db *gorm.DB) *UserRepositoryImpl {
+	return &UserRepositoryImpl{db: db}
 }
 
 // IsEmailExists checks if the email already exists in the database
+// @Summary Check if the email exists
+// @Description Checks if the provided email already exists in the database
+// @Param email query string true "Email address to check"
+// @Success 200 {string} string "true if email exists, false otherwise"
+
```

Comment:
no need to add swagger like documentation for all function, used just one liner description if needed.